GpsDataAnalyzer/calculator/deviation_calculator.py

```diff
@@ -0,0 +1,156 @@
+    offset_time_point_mapping: Dictionary, {DateTime: {"set1": GpsData, "set2": GpsData}, ...}
+    deviations_dataframe: Pandas Dataframe that holds values after calculation
+  """
+  def __init__(self, data_set_1, data_set_2):
+    self.data_set_1 = data_set_1
+    self.data_set_2 = data_set_2
+    self.starting_time_1 = None
+    self.starting_time_2 = None
+    self.offset_time_point_mapping = {}
+    self.deviations_dataframe = None
+
+    # comparing both algorithms for deciding offset, will delete this one post-testing
+    start = time.perf_counter()
+    print("Unoptimized lineup implementation:")
+    self.starting_time_1, self.starting_time_2 = alignment_algorithms.find_lineup_no_optimization(self.data_set_1,
+                                                                    self.data_set_2)
+    end = time.perf_counter()
+    print(f"Lined up data in {end - start:0.4f} seconds")
+    print("start time 1: " + str(self.starting_time_1))
+    print("start time 2: " + str(self.starting_time_2))
+    print("\n")
+
+    start = time.perf_counter()
+    print("Optimized lineup implementation:")
+    self.starting_time_1, self.starting_time_2 = alignment_algorithms.find_lineup(self.data_set_1,
+                                                                                  self.data_set_2)
+    end = time.perf_counter()
+    print(f"Lined up data in {end - start:0.4f} seconds")
+    print("start time 1: " + str(self.starting_time_1))
+    print("start time 2: " + str(self.starting_time_2))
+    print("\n")
+
+    if not self.starting_time_1 and not self.starting_time_2:
+      self.offset_time_point_mapping = self.create_time_to_point_mapping(0, 0)
+    elif self.data_set_1.gps_data_list[0].time > self.data_set_2.gps_data_list[0].time:
+      offset = (self.starting_time_1-self.starting_time_2).total_seconds()
+      self.offset_time_point_mapping = self.create_time_to_point_mapping(0, offset)
+    else:
+      offset = (self.starting_time_2-self.starting_time_1).total_seconds()
+      self.offset_time_point_mapping = self.create_time_to_point_mapping(offset, 0)
+
+  def get_deviation_dataframe(self):
+    """
+    Extracts and returns deviation for each valid timestamp & other information.
+
+    Returns:
+      A pandas dataframe including the shared timestamp with the offset included,
+      the deviations of lat/lon, the difference in speed, the difference in
+      altitude, and the original timestamps for each set
+    """
+    if self.deviations_dataframe: 
+      return self.deviations_dataframe
+    else:
+      time_list, deviation_list, speed_differentials, altitude_differentials = [], [], [], []
+      set1_time_list, set2_time_list = [], []
+      time_point_mapping = self.offset_time_point_mapping
+
+      for timestamp in time_point_mapping:
+        entry = time_point_mapping[timestamp]
+        if len(entry) == 2:
+          time_list.append(timestamp)
```

Comment:
Can we replace it with `time_list.append(timestamp.time)`? It's too long to show the whole timestamp on the graph.